atef/check.py

```diff
@@ -308,9 +465,39 @@ async def get_data_for_signal_async(
             executor=executor,
         )
 
+    def prepare(self, cache: Optional[DataCache] = None) -> None:
```

Comment:
Does it matter that this is a function here, but is a coroutine in all the inheriting classes?

Reply:
You're right, it should be a coroutine.  We never run into problems with it because it's re-defined immediately.